tests/qadence/test_dagger.py

```diff
@@ -147,3 +149,9 @@ def test_all_self_adjoint_blocks(block: Tuple[int, AbstractBlock]) -> None:
 )
 def test_composite_blocks_no_fails(block: AbstractBlock) -> None:
     assert isinstance(block.dagger(), AbstractBlock)
+
+
+@given(st.restricted_circuits())
+@settings(deadline=None)
+def test_circuit_dagger(circuit: QuantumCircuit) -> None:
+    circuit == circuit.dagger().dagger()
```

Comment:
I think there was a test somewhere that did not just do `.dagger().dagger()`, but explicitly checked a few (nested) circuits. could we get that back? here the `dagger()` could also just be a no-op

Reply:
added explicit test